woocommerce/woocommerce-admin-test-helper
# Allow being able to add extendable task items for the task list

Fixes: #18 

~~This is an initial PR for being able to add extendable task items for the task list.
Currently I only have it set to a specific number, and it will auto generate the number of task items, it would be nice to change it so we can add all the task lists manually, having control over (title, dismisable, completed, etc...) (created an issue for it -> https://github.com/woocommerce/woocommerce-admin-test-helper/issues/18).~~

There are two main changes in this PR:
- I updated the script enqueue logic to only load the app on the **Tools > WCA Test Helper** page and adding a separate bundle used for the filters that gets enqueued on every admin page.
- I created a extended task list tab, with a list of test helper tasks, someone can also create new tasks with a couple different options as shown in the GIF below. Note: The task list type can either be `setup` or `extension`, if set to `setup` it will show in the main task list, if set to `extension` it is shown as an extension task list (also shown in the GIF).

![admin-test-helper-task-list](https://user-images.githubusercontent.com/2240960/122761098-3e676400-d272-11eb-8e7b-a4a513abe83c.gif)


### Testing instructions

- Activate the WC Admin Test Helper plugin with this branch loaded
- Go to **Tools > WCA Test Helper > Extended Tasks**
- Create a couple tasks, one of each type
- Navigate to **WooCommerce > Home**
- Newly created tasks should be displayed (if task list is enabled) you should be able to click on them and complete them.
- Navigate back to **Tools > WCA Test Helper > Extended Tasks** and edit and or delete a task
- Go back to **WooCommerce > Home** the tasks should not be displayed if deleted or correctly updated if edited (can only edit is completed).
